### PR TITLE
251 lane route ticket history replace printJS with reprintTicket mutation

### PR DESCRIPTION
**Pull request by HaleyBoles**
**!! IMPORTANT !!**
Added react-PDF package, will need to run npm install
https://github.com/wojtekmaj/react-pdf

**Ticket Reprint Component**
- New!
- Pop-up to show PDF preview (a la Windows) via react-PDF 
- Uses ticket's lane's default printer as default, but does show all printers
- use reprintTicket mutation instead of printJS

**Queries & Mutations**
- Added Ticket_Blob_Data query to get just base ticket info and the blob stuff (image, pdf, etc)
- Added Reprint_Ticket mutation

**Ticket History**
- Replaced printJS with Ticket Reprint Component
- Replaced Ticket_History query with Ticket_Blob_Data query
- Cleanup variable/function typing 
    - Use fragment's type instead of gql table's type
- Cleanup grid cell renderers/buttons/popups
    - Combined most popup variables etc before I broke out reprint but it was nice so why undo it

**Lane Route** 
- Replaced printJS with Ticket Reprint Component
- Replaced Ticket_History query with Ticket_Blob_Data query

**Popup Component**
- Had to do some cleanup for Ticket Reprint
    - added a max-height to the main data container
    - added option to pass any string to max width prop in case you want to use something like viewport width instead of static max width

**Trucks, Requests, Unsynced**
- Removed max-h that became redundant with Popup changes

**Utils**
- Moved getBase64Url function to Utils for re-use

**Text Area**
- Added full width prop

### CODE REVIEW

**Review by sspoon865 — CHANGES_REQUESTED** [review]
See comments

File: src/routes/TicketHistory.tsx [hunk 1]
```diff
@@ -38,17 +41,20 @@ import { ROUTES } from "@/routes";
...
 
 	const navigate = useNavigate();
+	const queryClient = useQueryClient();
 
 	//Setting up default to and from dates
 	const todaysDate = new Date();
 	// Can use daysBack in the future for a default range
-	const daysBack = 0;
+	//TODO: reset to 0
+	const daysBack = 5;
```
[the remark this answers is not in the document]
Comment (HaleyBoles) [reply src/routes/TicketHistory.tsx:57]: oopsies 😂

File: src/components/TicketReprint.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	}, [props.isOpen]);
+
+	return (
+		<Popup
+			title={"Reprint Ticket #" + props.ticket?.ticketNumber}
+			maxSize="max-w-[75vw]"
+			contentClassName="h-full"
+			isOpen={props.isOpen}
+			onClose={props.onClose}
+			content={
```
[the remark this answers is not in the document]
Comment (HaleyBoles) [reply src/components/TicketReprint.tsx:56]: Agreed! I left the box open for now so you could do multiple copies. If we're wanting multiple copies option, I can go ahead and auto-close for now then it won't even matter when we can do more than one at once.

[the remark this answers is not in the document]
Comment (HaleyBoles) [reply src/components/TicketReprint.tsx:56]: https://github.com/TAC-Insight/fw-scale-backend/issues/36

**Review by sspoon865 — APPROVED** [review]

**Pull request merged**